pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -178,3 +180,82 @@ func (i *deploymentClientImpl) listDeploymentsForConfig(namespace, configName st
 func (i *deploymentClientImpl) updateDeployment(namespace string, deployment *kapi.ReplicationController) (*kapi.ReplicationController, error) {
 	return i.updateDeploymentFunc(namespace, deployment)
 }
+
+// findDetails inspects the given deployment configuration for any failure causes
+// and returns any found details about it
+func (c *DeploymentConfigController) findDetails(config *deployapi.DeploymentConfig) (string, *kapi.ReplicationControllerList, bool, error) {
+	// Check if any existing inflight deployments (any non-terminal state).
+	existingDeployments, err := c.deploymentClient.listDeploymentsForConfig(config.Namespace, config.Name)
+	if err != nil {
+		return "", nil, false, fmt.Errorf("couldn't list deployments for deploymentConfig %s: %v", deployutil.LabelForDeploymentConfig(config), err)
+	}
+	// check if the latest deployment exists
+	// we'll return after we've dealt with the multiple-active-deployments case
+	latestDeploymentExists, latestDeploymentStatus := deployutil.LatestDeploymentInfo(config, existingDeployments)
+	if latestDeploymentExists && latestDeploymentStatus == deployapi.DeploymentStatusComplete {
+		// If the latest deployment exists and is complete, clear the dc message
+		return "", existingDeployments, latestDeploymentExists, nil
+	}
+
+	// Rest of the code will handle non-existing or failed latest deployment causes
+	// TODO: Inspect pod logs in case of failed latest
+
+	// If the configuration we are inspecting doesn't have a config change trigger
+	// there is no reason in proceeding to inspect its image change trigger
+	if !deployutil.HasChangeTrigger(config) {
+		return "", existingDeployments, latestDeploymentExists, nil
```

Comment:
I don't think this is right: if you have only an image trigger, and the image changes, the version gets bumped, and a new deployment happens.